azkaban-exec-server/src/test/java/azkaban/execapp/JobRunnerTest.java

```diff
@@ -297,15 +299,18 @@ public void testDelayedExecutionCancelledJob() throws InterruptedException {
         node.getStatus() == Status.KILLED);
     Assert.assertTrue(node.getStartTime() > 0 && node.getEndTime() > 0);
     Assert.assertTrue(node.getEndTime() - node.getStartTime() < 1000);
-    Assert.assertTrue(node.getStartTime() - startTime >= 2000);
-    Assert.assertTrue(node.getStartTime() - startTime <= 5000);
+    Assert.assertTrue(node.getStartTime() - startTime >= 1000);
+    Assert.assertTrue(node.getStartTime() - startTime <= 4000);
     Assert.assertTrue(runner.isKilled());
 
     final File logFile = new File(runner.getLogFilePath());
     final Props outputProps = runner.getNode().getOutputProps();
     Assert.assertTrue(outputProps == null);
     Assert.assertTrue(logFile.exists());
 
+    // sleep so that there's time to make the "DB update" for KILLED status
+    Thread.sleep(1000L);
```

Comment:
An extra one second delay in test is not ideal. It would be great if you can think of ways to make it better.